internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/compute_tf"
+	"github.com/databricks/terraform-provider-databricks/libraries"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/databricks-sdk-go"
+)
+
+const libraryDefaultInstallationTimeout = 15 * time.Minute
+
+var _ resource.ResourceWithConfigure = &LibraryResource{}
+
+func ResourceLibrary() resource.Resource {
+	return &LibraryResource{}
+}
+
+func parseId(id string) (string, string) {
+	split := strings.SplitN(id, "/", 2)
+	if len(split) != 2 {
+		return "unknown", "unknown"
+	}
```

Comment:
I know you're porting this directly from the old implementation, but this is just gross...

Reply:
Updated it into a panic, not sure if that sounds good